src/app/services/shared.service.ts

```diff
@@ -45,11 +54,31 @@ export class SharedService {
         this.toastrService.warning(message.description, message.title);
     }
 
+    public set status(value: ConnectivityStatus) {
+        this._status = value;
+    }
+
+    public get status(): ConnectivityStatus {
+        return this._status;
+    }
+
     requestStarted() {
         this.requestsSubject.next(+1);
     }
 
     requestFinished() {
         this.requestsSubject.next(-1);
     }
+
+    setStackTrace(stackTrace: Error | null) {
```

Comment:
Same setter/getter comment

Reply:
I will still keep the `getStackTrace` mostly because it has some minor logic.